packages/core/src/BlockNoteEditor.ts

```diff
@@ -217,25 +215,24 @@ export class BlockNoteEditor<BSchema extends BlockSchema = DefaultBlockSchema> {
 
     this.schema = newOptions.blockSchema;
 
-    this.imageUpload = async (file) => {
-      if (newOptions.uploadImage) {
-        return newOptions.uploadImage(file);
-      }
-
-      // TODO: Proper backend - right now using imgbb.com since you can get an
-      //  API key for free by just making an account.
-      //  https://imgbb.com/
-      const body = new FormData();
-      body.append("key", IMAGE_UPLOAD_API_KEY);
-      body.append("image", file);
-
-      return fetch("https://api.imgbb.com/1/upload?expiration=600", {
-        method: "POST",
-        body: body,
-      })
-        .then((response) => response.json())
-        .then((data) => data.data.url);
-    };
+    this.uploadFile =
+      newOptions.uploadFile ||
+      (async (file) => {
```

Comment:
I think we should move this example upload function to a separate file, and name it sth like `uploadToTmpFilesOrg_DEV_ONLY`. Then, it becomes clear that this should be replaced by the user. I also don't think we should set it as a default, but that the user should always pass `uploadFile` as part of the options.

Reply:
Instead of a separate file, I think it's better to just add it as an example to the docs

Reply:
Hmm, then the code to get started (both for dev and in docs) will be quite cumbersome, because they’ll always need to cooy and paste that code